File: app/src/main/java/com/shar2wy/moviesapp/models/moviesRepo/MovieRemoteDataSource.java
package com.shar2wy.moviesapp.models.moviesRepo;

import android.content.Context;

import com.shar2wy.moviesapp.network.ApiClient;
import com.shar2wy.moviesapp.network.ApiService;

import java.util.List;

import io.reactivex.Flowable;
import io.reactivex.android.schedulers.AndroidSchedulers;

/**
 * Created by shar2wy
 * on 4/4/18.
 * Description: description goes here
 */
class MovieRemoteDataSource {

    MovieLocalDataSource mMovieLocalDataSource;

    ApiService apiService;
    String apiKey = "38db4ee0b3b9fe63c95b9c835b73021f";

    public MovieRemoteDataSource(Context context) {
        apiService = ApiClient.getClient(context).create(ApiService.class);
        mMovieLocalDataSource = new MovieLocalDataSource();
    }

    public Flowable<List<Movie>> getMovies() {
        return apiService.getTopRatedMovies(apiKey)
                .map(moviesResponse -> moviesResponse.getResults())
                .doOnNext(movies -> {
                    //saveToRealm
//                        mMovieLocalDataSource.saveMovies(movies);
                }).subscribeOn(AndroidSchedulers.mainThread());
    }
}

File: app/src/main/java/com/shar2wy/moviesapp/models/moviesRepo/MovieRemoteDataSource.kt
package com.shar2wy.moviesapp.models.moviesRepo

import com.shar2wy.moviesapp.network.ApiClient
import com.shar2wy.moviesapp.network.ApiService
import com.shar2wy.moviesapp.util.Constants.apiKey
import io.reactivex.Flowable

/**
 * Created by shar2wy
 * on 4/4/18.
 * Description: description goes here
 */
internal class MovieRemoteDataSource() {

    var mMovieLocalDataSource: MovieLocalDataSource = MovieLocalDataSource()

    var apiService: ApiService = ApiClient.getInstance().getService(ApiService::class.java)

    val movies: Flowable<List<Movie>>
        get() = apiService.getTopRatedMovies(apiKey)
                .map { moviesResponse -> moviesResponse.results }
                .doOnNext { movies -> }
}
